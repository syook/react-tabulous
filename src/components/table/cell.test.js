import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import TableCell from './cell';

configure({ adapter: new Adapter() });

const data = [
  { Description: 'Dev', Name: 'David', id: 1, objIndex: 0 },
  { Description: 'QA', Name: 'John', id: 2, objIndex: 1 },
];

describe('TableCell', () => {
  it('should render without crashing', () => {
    shallow(<TableCell data={{ data }} row={{ objIndex: 0 }} />);
  });

  it('should match snapshot', () => {
    const tree = mount(<TableCell data={{ data }} row={{ objIndex: 0 }} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should match snapshot if currentItem is not found', () => {
    const tree = mount(<TableCell data={{ data }} row={{ objIndex: 3 }} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should match snapshot if currentItem is found', () => {
    const tree = mount(<TableCell data={{ data }} row={{ objIndex: 1 }} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
