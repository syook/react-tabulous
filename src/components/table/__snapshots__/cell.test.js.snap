// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableCell should match snapshot 1`] = `
<TableCell
  data={
    Object {
      "data": Array [
        Object {
          "Description": "Dev",
          "Name": "David",
          "id": 1,
          "objIndex": 0,
        },
        Object {
          "Description": "QA",
          "Name": "John",
          "id": 2,
          "objIndex": 1,
        },
      ],
    }
  }
  row={
    Object {
      "objIndex": 0,
    }
  }
/>
`;

exports[`TableCell should match snapshot if currentItem is found 1`] = `
<TableCell
  data={
    Object {
      "data": Array [
        Object {
          "Description": "Dev",
          "Name": "David",
          "id": 1,
          "objIndex": 0,
        },
        Object {
          "Description": "QA",
          "Name": "John",
          "id": 2,
          "objIndex": 1,
        },
      ],
    }
  }
  row={
    Object {
      "objIndex": 1,
    }
  }
/>
`;

exports[`TableCell should match snapshot if currentItem is not found 1`] = `
<TableCell
  data={
    Object {
      "data": Array [
        Object {
          "Description": "Dev",
          "Name": "David",
          "id": 1,
          "objIndex": 0,
        },
        Object {
          "Description": "QA",
          "Name": "John",
          "id": 2,
          "objIndex": 1,
        },
      ],
    }
  }
  row={
    Object {
      "objIndex": 3,
    }
  }
/>
`;
